diff -rupN linux-ba5b12366970b0fb07c29217a44334fdab94ea81/drivers/net/wireless/ath/ath9k/common-init.c linux-ba5b12366970b0fb07c29217a44334fdab94ea81-new/drivers/net/wireless/ath/ath9k/common-init.c
--- linux-ba5b12366970b0fb07c29217a44334fdab94ea81/drivers/net/wireless/ath/ath9k/common-init.c	2015-01-14 16:55:55.000000000 +0000
+++ linux-ba5b12366970b0fb07c29217a44334fdab94ea81-new/drivers/net/wireless/ath/ath9k/common-init.c	2015-02-06 11:47:36.189860919 +0000
@@ -22,7 +22,7 @@
 	.band = IEEE80211_BAND_2GHZ, \
 	.center_freq = (_freq), \
 	.hw_value = (_idx), \
-	.max_power = 20, \
+	.max_power = 30, \
 }
 
 #define CHAN5G(_freq, _idx) { \
diff -rupN linux-ba5b12366970b0fb07c29217a44334fdab94ea81/drivers/net/wireless/ath/regd.c linux-ba5b12366970b0fb07c29217a44334fdab94ea81-new/drivers/net/wireless/ath/regd.c
--- linux-ba5b12366970b0fb07c29217a44334fdab94ea81/drivers/net/wireless/ath/regd.c	2015-01-14 16:55:55.000000000 +0000
+++ linux-ba5b12366970b0fb07c29217a44334fdab94ea81-new/drivers/net/wireless/ath/regd.c	2015-02-06 11:48:56.033763008 +0000
@@ -33,7 +33,7 @@ static int __ath_regd_init(struct ath_re
  */
 
 /* Only these channels all allow active scan on all world regulatory domains */
-#define ATH9K_2GHZ_CH01_11	REG_RULE(2412-10, 2462+10, 40, 0, 20, 0)
+#define ATH9K_2GHZ_CH01_11	REG_RULE(2412-10, 2462+10, 40, 0, 30, 0)
 
 /* We enable active scan on these a case by case basis by regulatory domain */
 #define ATH9K_2GHZ_CH12_13	REG_RULE(2467-10, 2472+10, 40, 0, 20,\
