diff -rupN ../tmp/buildroot-2014.02/package/Config.in package/Config.in
--- ../tmp/buildroot-2014.02/package/Config.in	2014-02-27 20:51:23.000000000 +0000
+++ package/Config.in	2014-04-04 23:54:17.733671580 +0100
@@ -2,6 +2,12 @@ menu "Target packages"
 
 source "package/busybox/Config.in"
 
+menu "Rpicopter"
+source "package/libnih/Config.in"
+source "package/upstart/Config.in"
+source "package/rpicopter/Config.in"
+endmenu
+
 menu "Audio and video applications"
 source "package/alsa-utils/Config.in"
 source "package/aumix/Config.in"
diff -rupN ../tmp/buildroot-2014.02/package/libnih/Config.in package/libnih/Config.in
--- ../tmp/buildroot-2014.02/package/libnih/Config.in	1970-01-01 01:00:00.000000000 +0100
+++ package/libnih/Config.in	2014-04-04 23:54:44.530389051 +0100
@@ -0,0 +1,9 @@
+config BR2_PACKAGE_LIBNIH
+    bool "libnih"
+    select BR2_PACKAGE_DBUS
+    select BR2_PACKAGE_EXPAT
+    select BR2_PACKAGE_PKGCONF
+    help
+        libnih is a small library for C application development
+        containing functions that, despite its name, are not 
+        implemented elsewhere in the standard library set.
diff -rupN ../tmp/buildroot-2014.02/package/libnih/libnih.mk package/libnih/libnih.mk
--- ../tmp/buildroot-2014.02/package/libnih/libnih.mk	1970-01-01 01:00:00.000000000 +0100
+++ package/libnih/libnih.mk	2014-04-04 23:53:56.477102575 +0100
@@ -0,0 +1,26 @@
+LIBNIH_VERSION = 1.0.3
+LIBNIH_SITE = https://launchpad.net/libnih/1.0/$(LIBNIH_VERSION)/+download
+LIBNIH_SOURCE = libnih-$(LIBNIH_VERSION).tar.gz
+HOST_LIBNIH_DEPENDENCIES = host-dbus host-expat
+LIBNIH_DEPENDENCIES = dbus expat pkgconf host-libnih
+LIBNIH_INSTALL_STAGING = YES
+
+HOST_LIBNIH_AUTORECONF = YES
+
+define HOST_LIBNIH_CONFIG_CMDS
+	(cd $(@D) && rm -rf config.cache; \
+	        $(HOST_CONFIGURE_OPTS) \
+		CFLAGS="$(HOST_CFLAGS)" \
+		LDFLAGS="$(HOST_LDFLAGS)" \
+		./configure \
+		--prefix="$(HOST_DIR)/usr" \
+		--sysconfdir="$(HOST_DIR)/etc" \
+	)
+endef
+
+LIBNIH_AUTORECONF = YES
+
+LIBNIH_CONF_OPT += NIH_DBUS_TOOL=$(HOST_DIR)/usr/bin/nih-dbus-tool
+
+$(eval $(autotools-package))
+$(eval $(host-autotools-package))
diff -rupN ../tmp/buildroot-2014.02/package/rpicopter/Config.in package/rpicopter/Config.in
--- ../tmp/buildroot-2014.02/package/rpicopter/Config.in	1970-01-01 01:00:00.000000000 +0100
+++ package/rpicopter/Config.in	2014-04-05 22:45:48.061677382 +0100
@@ -0,0 +1,13 @@
+config BR2_PACKAGE_RPICOPTER
+	bool "rpicopter"
+	help
+	  Adds rpicopter utils package
+	  https://github.com/rpicopter 
+
+if BR2_PACKAGE_RPICOPTER
+source "package/rpicopter/rpic-bluez-tools/Config.in"
+source "package/rpicopter/rpic-utils/Config.in"
+source "package/rpicopter/rpic-pigpio/Config.in"
+source "package/rpicopter/rpic-servoblaster/Config.in"
+source "package/rpicopter/rpic-qtsixa/Config.in"
+endif
diff -rupN ../tmp/buildroot-2014.02/package/rpicopter/rpic-bluez-tools/Config.in package/rpicopter/rpic-bluez-tools/Config.in
--- ../tmp/buildroot-2014.02/package/rpicopter/rpic-bluez-tools/Config.in	1970-01-01 01:00:00.000000000 +0100
+++ package/rpicopter/rpic-bluez-tools/Config.in	2014-04-05 23:00:59.734985955 +0100
@@ -0,0 +1,7 @@
+config BR2_PACKAGE_RPIC_BLUEZ_TOOLS
+	bool "rpic-bluez-tools"
+    depends on dbus-glib
+    depends on readline
+	help
+      bluez-tools
+	  https://github.com/rpicopter/bluez-tools
diff -rupN ../tmp/buildroot-2014.02/package/rpicopter/rpic-bluez-tools/rpic-bluez-tools.mk package/rpicopter/rpic-bluez-tools/rpic-bluez-tools.mk
--- ../tmp/buildroot-2014.02/package/rpicopter/rpic-bluez-tools/rpic-bluez-tools.mk	1970-01-01 01:00:00.000000000 +0100
+++ package/rpicopter/rpic-bluez-tools/rpic-bluez-tools.mk	2014-04-05 22:52:01.300046369 +0100
@@ -0,0 +1,13 @@
+#############################################################
+#
+# rpic-bluez-tools 
+#
+#############################################################
+RPIC_BLUEZ_TOOLS_VERSION = master
+RPIC_BLUEZ_TOOLS_SITE_METHOD = git
+RPIC_BLUEZ_TOOLS_REPO_URL = https://github.com/rpicopter/bluez-tools.git
+RPIC_BLUEZ_TOOLS_SITE = $(call qstrip,$(RPIC_BLUEZ_TOOLS_REPO_URL))
+RPIC_BLUEZ_TOOLS_INSTALL_TARGET = YES
+RPIC_BLUEZ_TOOLS_AUTORECONF = YES
+
+$(eval $(autotools-package))
diff -rupN ../tmp/buildroot-2014.02/package/rpicopter/rpicopter.mk package/rpicopter/rpicopter.mk
--- ../tmp/buildroot-2014.02/package/rpicopter/rpicopter.mk	1970-01-01 01:00:00.000000000 +0100
+++ package/rpicopter/rpicopter.mk	2014-04-04 23:07:37.362969244 +0100
@@ -0,0 +1 @@
+include package/rpicopter/*/*.mk
diff -rupN ../tmp/buildroot-2014.02/package/rpicopter/rpic-pigpio/Config.in package/rpicopter/rpic-pigpio/Config.in
--- ../tmp/buildroot-2014.02/package/rpicopter/rpic-pigpio/Config.in	1970-01-01 01:00:00.000000000 +0100
+++ package/rpicopter/rpic-pigpio/Config.in	2014-04-04 23:07:37.362969244 +0100
@@ -0,0 +1,5 @@
+config BR2_PACKAGE_RPIC_PIGPIO
+	bool "rpic-pigpio"
+	help
+	  Adds pigpio package
+	  https://github.com/rpicopter/pigpio
diff -rupN ../tmp/buildroot-2014.02/package/rpicopter/rpic-pigpio/rpic-pigpio.mk package/rpicopter/rpic-pigpio/rpic-pigpio.mk
--- ../tmp/buildroot-2014.02/package/rpicopter/rpic-pigpio/rpic-pigpio.mk	1970-01-01 01:00:00.000000000 +0100
+++ package/rpicopter/rpic-pigpio/rpic-pigpio.mk	2014-04-04 23:07:37.362969244 +0100
@@ -0,0 +1,34 @@
+#############################################################
+#
+# rpic-pigpio 
+#
+#############################################################
+#RPIC_PIGPIO_VERSION = a26ed773d7792cca7f67c7baca530940cb991e5d
+RPIC_PIGPIO_VERSION = master
+RPIC_PIGPIO_SITE_METHOD = git
+RPIC_PIGPIO_REPO_URL = https://github.com/rpicopter/pigpio.git
+RPIC_PIGPIO_SITE = $(call qstrip,$(RPIC_PIGPIO_REPO_URL))
+#PIGPIO_INSTALL_STAGING = YES
+
+#$(MAKE) CC=$(TARGET_CC) LD=$(TARGET_LD) -C $(@D) all
+
+define RPIC_PIGPIO_BUILD_CMDS
+ $(MAKE) $(TARGET_CONFIGURE_OPTS) -C $(@D) all
+endef
+
+define RPIC_PIGPIO_INSTALL_TARGET_CMDS
+	$(INSTALL) -D -m 0755 -d              $(TARGET_DIR)/opt/pigpio/cgi
+	$(INSTALL)  -m 0755 -d              $(TARGET_DIR)/usr/local/include
+	$(INSTALL)  -m 0644 $(@D)/pigpio.h        $(TARGET_DIR)/usr/local/include
+	$(INSTALL) -m 0644 $(@D)/pigpiod_if.h    $(TARGET_DIR)/usr/local/include
+	$(INSTALL) -m 0755 -d              $(TARGET_DIR)/usr/local/lib
+	$(INSTALL) -m 0644 $(@D)/libpigpio.a     $(TARGET_DIR)/usr/local/lib
+	$(INSTALL) -m 0644 $(@D)/libpigpiod_if.a $(TARGET_DIR)/usr/local/lib
+	$(INSTALL) -m 0755 -d              $(TARGET_DIR)/usr/local/bin
+	$(INSTALL) -m 0755 $(@D)/pig2vcd         $(TARGET_DIR)/usr/local/bin
+	$(INSTALL)  -m 0755 $(@D)/pigpiod        $(TARGET_DIR)/usr/local/bin
+	$(INSTALL)  -m 0755 $(@D)/pigs           $(TARGET_DIR)/usr/local/bin
+
+endef
+
+$(eval $(generic-package))
diff -rupN ../tmp/buildroot-2014.02/package/rpicopter/rpic-qtsixa/Config.in package/rpicopter/rpic-qtsixa/Config.in
--- ../tmp/buildroot-2014.02/package/rpicopter/rpic-qtsixa/Config.in	1970-01-01 01:00:00.000000000 +0100
+++ package/rpicopter/rpic-qtsixa/Config.in	2014-04-04 23:07:37.362969244 +0100
@@ -0,0 +1,6 @@
+config BR2_PACKAGE_RPIC_QTSIXA
+	bool "rpic-qtsixa"
+	depends on BR2_PACKAGE_LIBUSB_COMPAT
+	help
+	  Adds support for PS3 SixAxis controller
+	  https://github.com/rpicopter/QtSixA 
diff -rupN ../tmp/buildroot-2014.02/package/rpicopter/rpic-qtsixa/rpic-qtsixa.mk package/rpicopter/rpic-qtsixa/rpic-qtsixa.mk
--- ../tmp/buildroot-2014.02/package/rpicopter/rpic-qtsixa/rpic-qtsixa.mk	1970-01-01 01:00:00.000000000 +0100
+++ package/rpicopter/rpic-qtsixa/rpic-qtsixa.mk	2014-04-05 23:31:14.021069256 +0100
@@ -0,0 +1,44 @@
+#############################################################
+#
+# rpic-qtsixa
+#
+#############################################################
+#QTSIXA_VERSION = ac0a6c3ef460acebd8f60fa9a44946056deed0aa
+RPIC_QTSIXA_VERSION = master
+RPIC_QTSIXA_SITE_METHOD = git
+RPIC_QTSIXA_REPO_URL = https://github.com/rpicopter/QtSixA.git
+RPIC_QTSIXA_SITE = $(call qstrip,$(RPIC_QTSIXA_REPO_URL))
+
+define RPIC_QTSIXA_BUILD_CMDS
+ $(MAKE) $(TARGET_CONFIGURE_OPTS) -C $(@D)/sixad
+ $(MAKE) $(TARGET_CONFIGURE_OPTS) -C $(@D)/utils
+endef
+
+define RPIC_QTSIXA_INSTALL_TARGET_CMDS
+       $(INSTALL) -m 0755 -d              $(TARGET_DIR)/usr/bin
+       $(INSTALL) -m 0755 -d              $(TARGET_DIR)/usr/sbin
+       $(INSTALL) -m 0755 -d              $(TARGET_DIR)/etc/default
+       $(INSTALL) -m 0755 -d              $(TARGET_DIR)/etc/init.d
+       $(INSTALL) -m 0755 -d              $(TARGET_DIR)/etc/logrotate.d
+       $(INSTALL) -m 0755 -d              $(TARGET_DIR)/var/lib/sixad
+       $(INSTALL) -m 0755 -d              $(TARGET_DIR)/var/lib/sixad/profiles
+
+	$(INSTALL) -m 644 $(@D)/sixad/sixad.default $(TARGET_DIR)/etc/default/sixad
+	$(INSTALL) -m 644 $(@D)/sixad/sixad.default $(TARGET_DIR)/etc/init.d/sixad
+	$(INSTALL) -m 644 $(@D)/sixad/sixad.log $(TARGET_DIR)/etc/logrotate.d/sixad
+	$(INSTALL) -m 755 $(@D)/sixad/sixad $(TARGET_DIR)/usr/bin/
+	$(INSTALL) -m 755 $(@D)/sixad/bins/sixad-bin $(TARGET_DIR)/usr/sbin/
+	$(INSTALL) -m 755 $(@D)/sixad/bins/sixad-sixaxis $(TARGET_DIR)/usr/sbin/
+	$(INSTALL) -m 755 $(@D)/sixad/bins/sixad-remote $(TARGET_DIR)/usr/sbin/
+	$(INSTALL) -m 755 $(@D)/sixad/bins/sixad-3in1 $(TARGET_DIR)/usr/sbin/
+	$(INSTALL) -m 755 $(@D)/sixad/bins/sixad-raw $(TARGET_DIR)/usr/sbin/
+	$(INSTALL) -m 755 $(@D)/sixad/sixad-dbus-blocker $(TARGET_DIR)/usr/sbin/
+	@chmod 777 -R $(TARGET_DIR)/var/lib/sixad/
+
+
+	$(INSTALL) -m 755 $(@D)/utils/bins/sixpair $(TARGET_DIR)/usr/bin/
+	$(INSTALL) -m 755 $(@D)/utils/bins/sixpair-kbd $(TARGET_DIR)/usr/bin/
+	$(INSTALL) -m 755 $(@D)/utils/bins/hidraw-dump $(TARGET_DIR)/usr/bin/
+endef
+
+$(eval $(generic-package))
diff -rupN ../tmp/buildroot-2014.02/package/rpicopter/rpic-servoblaster/Config.in package/rpicopter/rpic-servoblaster/Config.in
--- ../tmp/buildroot-2014.02/package/rpicopter/rpic-servoblaster/Config.in	1970-01-01 01:00:00.000000000 +0100
+++ package/rpicopter/rpic-servoblaster/Config.in	2014-04-04 23:07:37.362969244 +0100
@@ -0,0 +1,5 @@
+config BR2_PACKAGE_RPIC_SERVOBLASTER
+	bool "rpic-servoblaster"
+	help
+	  Adds servoblaster
+	  https://github.com/rpicopter/ServoBlaster
diff -rupN ../tmp/buildroot-2014.02/package/rpicopter/rpic-servoblaster/rpic-servoblaster.mk package/rpicopter/rpic-servoblaster/rpic-servoblaster.mk
--- ../tmp/buildroot-2014.02/package/rpicopter/rpic-servoblaster/rpic-servoblaster.mk	1970-01-01 01:00:00.000000000 +0100
+++ package/rpicopter/rpic-servoblaster/rpic-servoblaster.mk	2014-04-04 23:07:37.362969244 +0100
@@ -0,0 +1,21 @@
+#############################################################
+#
+# rpic-servoblaster
+#
+#############################################################
+#SERVOBLASTER_VERSION = ac0a6c3ef460acebd8f60fa9a44946056deed0aa
+RPIC_SERVOBLASTER_VERSION = master
+RPIC_SERVOBLASTER_SITE_METHOD = git
+RPIC_SERVOBLASTER_REPO_URL = https://github.com/rpicopter/servoblaster.git
+RPIC_SERVOBLASTER_SITE = $(call qstrip,$(RPIC_SERVOBLASTER_REPO_URL))
+
+define RPIC_SERVOBLASTER_BUILD_CMDS
+ $(MAKE) $(TARGET_CONFIGURE_OPTS) -C $(@D)/user all
+endef
+
+define RPIC_SERVOBLASTER_INSTALL_TARGET_CMDS
+       $(INSTALL) -m 0755 -d              $(TARGET_DIR)/usr/local/bin
+       $(INSTALL) -m 0755 $(@D)/user/servod         $(TARGET_DIR)/usr/local/bin
+endef
+
+$(eval $(generic-package))
diff -rupN ../tmp/buildroot-2014.02/package/rpicopter/rpic-utils/Config.in package/rpicopter/rpic-utils/Config.in
--- ../tmp/buildroot-2014.02/package/rpicopter/rpic-utils/Config.in	1970-01-01 01:00:00.000000000 +0100
+++ package/rpicopter/rpic-utils/Config.in	2014-04-04 23:07:37.366969350 +0100
@@ -0,0 +1,7 @@
+config BR2_PACKAGE_RPIC_UTILS
+	bool "rpic-utils"
+#	select BR2_PACKAGE_BLUEZ_UTILS_COMPAT
+#	select BR2_PACKAGE_UDEV
+	help
+	  Adds utils package
+	  https://github.com/rpicopter/utils
diff -rupN ../tmp/buildroot-2014.02/package/rpicopter/rpic-utils/rpic-utils.mk package/rpicopter/rpic-utils/rpic-utils.mk
--- ../tmp/buildroot-2014.02/package/rpicopter/rpic-utils/rpic-utils.mk	1970-01-01 01:00:00.000000000 +0100
+++ package/rpicopter/rpic-utils/rpic-utils.mk	2014-04-04 23:07:37.366969350 +0100
@@ -0,0 +1,31 @@
+#############################################################
+#
+# rpic-utils 
+#
+#############################################################
+RPIC_UTILS_VERSION = master
+RPIC_UTILS_SITE_METHOD = git
+RPIC_UTILS_REPO_URL = https://github.com/rpicopter/utils.git
+RPIC_UTILS_SITE = $(call qstrip,$(RPIC_UTILS_REPO_URL))
+#PIGPIO_INSTALL_STAGING = YES
+
+#$(MAKE) CC=$(TARGET_CC) LD=$(TARGET_LD) -C $(@D) all
+
+define RPIC_UTILS_BUILD_CMDS
+endef
+
+define RPIC_UTILS_INSTALL_TARGET_CMDS
+	install -d $(TARGET_DIR)/etc/udev/rules.d
+	install -d $(TARGET_DIR)/usr/local/bin
+	install -d $(TARGET_DIR)/var/local
+	install -m 655 $(@D)/root/etc/udev/rules.d/ps3controller.rules $(TARGET_DIR)/etc/udev/rules.d/ps3controller.rules
+	install -m 655 $(@D)/root/etc/udev/rules.d/91-btremove.rules $(TARGET_DIR)/etc/udev/rules.d/91-btremove.rules
+	install -m 655 $(@D)/root/etc/udev/rules.d/91-btadd.rules $(TARGET_DIR)/etc/udev/rules.d/91-btadd.rules
+	install -m 755 $(@D)/root/usr/local/bin/udevrun.sh $(TARGET_DIR)/usr/local/bin/udevrun.sh
+	install -m 755 $(@D)/root/usr/local/bin/sixpair.sh $(TARGET_DIR)/usr/local/bin/sixpair.sh
+	install -m 755 $(@D)/root/usr/local/bin/pand.sh $(TARGET_DIR)/usr/local/bin/pand.sh
+	install -m 755 $(@D)/root/usr/local/bin/btstart.sh $(TARGET_DIR)/usr/local/bin/btstart.sh
+	install -m 755 $(@D)/root/usr/local/bin/btstop.sh $(TARGET_DIR)/usr/local/bin/btstop.sh
+endef
+
+$(eval $(generic-package))
diff -rupN ../tmp/buildroot-2014.02/package/upstart/Config.in package/upstart/Config.in
--- ../tmp/buildroot-2014.02/package/upstart/Config.in	1970-01-01 01:00:00.000000000 +0100
+++ package/upstart/Config.in	2014-04-04 23:07:37.366969350 +0100
@@ -0,0 +1,13 @@
+config BR2_PACKAGE_UPSTART
+	bool "upstart"
+	depends on BR2_ROOTFS_DEVICE_CREATION_DYNAMIC_UDEV
+	depends on BR2_INET_IPV6
+	depends on BR2_TOOLCHAIN_HAS_THREADS # dbus
+	select BR2_PACKAGE_DBUS
+	select BR2_PACKAGE_LIBNIH
+	select BR2_PACKAGE_JSON_C
+	help
+		Upstart is an event-based replacement for the /sbin/init
+		daemon which handles starting of tasks and services
+		during boot, stopping them during shutdown and supervising 
+		them while the system is running.
diff -rupN ../tmp/buildroot-2014.02/package/upstart/network.conf package/upstart/network.conf
--- ../tmp/buildroot-2014.02/package/upstart/network.conf	1970-01-01 01:00:00.000000000 +0100
+++ package/upstart/network.conf	2014-04-04 23:07:37.370969457 +0100
@@ -0,0 +1,15 @@
+# networking
+##
+
+description "starts the network"
+
+start on runlevel [2345]
+stop on runlevel [!2345]
+
+pre-start script
+	/sbin/ifup -a
+end script
+
+post-stop script
+	/sbin/ifdown -a
+end script
diff -rupN ../tmp/buildroot-2014.02/package/upstart/upstart.mk package/upstart/upstart.mk
--- ../tmp/buildroot-2014.02/package/upstart/upstart.mk	1970-01-01 01:00:00.000000000 +0100
+++ package/upstart/upstart.mk	2014-04-04 23:07:37.370969457 +0100
@@ -0,0 +1,22 @@
+UPSTART_VERSION = 1.12.1
+UPSTART_SITE = http://upstart.ubuntu.com/download/$(UPSTART_VERSION)
+UPSTART_SOURCE = upstart-$(UPSTART_VERSION).tar.gz
+UPSTART_INSTALL_STAGING = YES
+UPSTART_DEPENDENCIES = udev dbus libnih json-c
+UPSTART_POST_INSTALL_TARGET_HOOKS += UPSTART_REPLACE_INITSCRIPTS
+
+UPSTART_AUTORECONF = YES
+
+UPSTART_CONF_OPT += --exec-prefix=/
+
+define UPSTART_REPLACE_INITSCRIPTS
+	#replace sysinitv scripts with upstart ones
+	rm -f $(TARGET_DIR)/etc/init.d/S20urandom
+	rm -f $(TARGET_DIR)/etc/init.d/S40network
+	$(INSTALL) -m 0644 package/upstart/urandom.conf \
+		$(TARGET_DIR)/etc/init
+	$(INSTALL) -m 0644 package/upstart/network.conf \
+		$(TARGET_DIR)/etc/init
+endef
+
+$(eval $(autotools-package))
diff -rupN ../tmp/buildroot-2014.02/package/upstart/urandom.conf package/upstart/urandom.conf
--- ../tmp/buildroot-2014.02/package/upstart/urandom.conf	1970-01-01 01:00:00.000000000 +0100
+++ package/upstart/urandom.conf	2014-04-04 23:07:37.370969457 +0100
@@ -0,0 +1,22 @@
+# urandom
+##
+
+description "saves the random seed between reboots"
+
+start on runlevel [2345]
+stop on runlevel [!2345]
+
+pre-start script
+	if [ -f /etc/random-seed ]
+	then
+		cat /etc/random-seed >/dev/urandom
+	fi
+	rm -f /etc/random-seed
+	dd if=/dev/urandom of=/etc/random-seed count=1 \
+		>/dev/null 2>&1
+end script
+
+post-stop script
+	dd if=/dev/urandom of=/etc/random-seed count=1 \
+		>/dev/null 2>&1
+end script
