diff -rupN buildroot-2014.08.orig/package/Config.in buildroot-2014.08/package/Config.in
--- buildroot-2014.08.orig/package/Config.in	2014-09-01 12:20:56.000000000 +0100
+++ buildroot-2014.08/package/Config.in	2014-11-09 21:34:50.059501993 +0000
@@ -2,6 +2,10 @@ menu "Target packages"
 
 	source "package/busybox/Config.in"
 
+menu "Rpicopter"
+	source "package/rpicopter/Config.in"
+endmenu
+
 menu "Audio and video applications"
 	source "package/alsa-utils/Config.in"
 	source "package/aumix/Config.in"
diff -rupN buildroot-2014.08.orig/package/rpicopter/Config.in buildroot-2014.08/package/rpicopter/Config.in
--- buildroot-2014.08.orig/package/rpicopter/Config.in	1970-01-01 01:00:00.000000000 +0100
+++ buildroot-2014.08/package/rpicopter/Config.in	2014-11-09 21:34:50.575759992 +0000
@@ -0,0 +1,15 @@
+config BR2_PACKAGE_RPICOPTER
+	bool "rpicopter"
+	help
+	  Adds rpicopter utils package
+	  https://github.com/rpicopter 
+
+if BR2_PACKAGE_RPICOPTER
+source "package/rpicopter/rpic-controller/Config.in"
+source "package/rpicopter/rpic-avrcontroller/Config.in"
+source "package/rpicopter/rpic-bluez-tools/Config.in"
+source "package/rpicopter/rpic-utils/Config.in"
+source "package/rpicopter/rpic-pigpio/Config.in"
+source "package/rpicopter/rpic-servoblaster/Config.in"
+source "package/rpicopter/rpic-qtsixa/Config.in"
+endif
diff -rupN buildroot-2014.08.orig/package/rpicopter/rpic-avrcontroller/Config.in buildroot-2014.08/package/rpicopter/rpic-avrcontroller/Config.in
--- buildroot-2014.08.orig/package/rpicopter/rpic-avrcontroller/Config.in	1970-01-01 01:00:00.000000000 +0100
+++ buildroot-2014.08/package/rpicopter/rpic-avrcontroller/Config.in	2014-11-09 21:34:50.575759992 +0000
@@ -0,0 +1,5 @@
+config BR2_PACKAGE_RPIC_AVRCONTROLLER
+	bool "rpic-avrcontroller"
+	help
+	  Adds avr controller package
+	  https://github.com/rpicopter/avrcontroller
diff -rupN buildroot-2014.08.orig/package/rpicopter/rpic-avrcontroller/rpic-avrcontroller.mk buildroot-2014.08/package/rpicopter/rpic-avrcontroller/rpic-avrcontroller.mk
--- buildroot-2014.08.orig/package/rpicopter/rpic-avrcontroller/rpic-avrcontroller.mk	1970-01-01 01:00:00.000000000 +0100
+++ buildroot-2014.08/package/rpicopter/rpic-avrcontroller/rpic-avrcontroller.mk	2014-11-09 21:34:50.575759992 +0000
@@ -0,0 +1,22 @@
+#############################################################
+#
+# rpic-avrcontroller 
+#
+#############################################################
+RPIC_AVRCONTROLLER_VERSION = master
+RPIC_AVRCONTROLLER_SITE_METHOD = git
+RPIC_AVRCONTROLLER_REPO_URL = https://github.com/rpicopter/AvrMiniCopter.git
+RPIC_AVRCONTROLLER_SITE = $(call qstrip,$(RPIC_AVRCONTROLLER_REPO_URL))
+#PIGPIO_INSTALL_STAGING = YES
+
+#$(MAKE) CC=$(TARGET_CC) LD=$(TARGET_LD) -C $(@D)/rpi all
+
+define RPIC_AVRCONTROLLER_BUILD_CMDS
+ $(MAKE) $(TARGET_CONFIGURE_OPTS) -C $(@D)/rpi all
+endef
+
+define RPIC_AVRCONTROLLER_INSTALL_TARGET_CMDS
+	DESTDIR=$(TARGET_DIR) $(MAKE) -C $(@D)/rpi install 
+endef
+
+$(eval $(generic-package))
diff -rupN buildroot-2014.08.orig/package/rpicopter/rpic-bluez-tools/Config.in buildroot-2014.08/package/rpicopter/rpic-bluez-tools/Config.in
--- buildroot-2014.08.orig/package/rpicopter/rpic-bluez-tools/Config.in	1970-01-01 01:00:00.000000000 +0100
+++ buildroot-2014.08/package/rpicopter/rpic-bluez-tools/Config.in	2014-11-09 21:34:50.575759992 +0000
@@ -0,0 +1,7 @@
+config BR2_PACKAGE_RPIC_BLUEZ_TOOLS
+    bool "rpic-bluez-tools"
+    depends on BR2_PACKAGE_DBUS_GLIB
+    depends on BR2_PACKAGE_READLINE
+    help
+        bluez-tools
+        https://github.com/rpicopter/bluez-tools
diff -rupN buildroot-2014.08.orig/package/rpicopter/rpic-bluez-tools/rpic-bluez-tools.mk buildroot-2014.08/package/rpicopter/rpic-bluez-tools/rpic-bluez-tools.mk
--- buildroot-2014.08.orig/package/rpicopter/rpic-bluez-tools/rpic-bluez-tools.mk	1970-01-01 01:00:00.000000000 +0100
+++ buildroot-2014.08/package/rpicopter/rpic-bluez-tools/rpic-bluez-tools.mk	2014-11-09 21:34:50.575759992 +0000
@@ -0,0 +1,13 @@
+#############################################################
+#
+# rpic-bluez-tools 
+#
+#############################################################
+RPIC_BLUEZ_TOOLS_VERSION = master
+RPIC_BLUEZ_TOOLS_SITE_METHOD = git
+RPIC_BLUEZ_TOOLS_REPO_URL = https://github.com/rpicopter/bluez-tools.git
+RPIC_BLUEZ_TOOLS_SITE = $(call qstrip,$(RPIC_BLUEZ_TOOLS_REPO_URL))
+RPIC_BLUEZ_TOOLS_INSTALL_TARGET = YES
+RPIC_BLUEZ_TOOLS_AUTORECONF = YES
+
+$(eval $(autotools-package))
diff -rupN buildroot-2014.08.orig/package/rpicopter/rpic-controller/Config.in buildroot-2014.08/package/rpicopter/rpic-controller/Config.in
--- buildroot-2014.08.orig/package/rpicopter/rpic-controller/Config.in	1970-01-01 01:00:00.000000000 +0100
+++ buildroot-2014.08/package/rpicopter/rpic-controller/Config.in	2014-11-09 21:34:50.575759992 +0000
@@ -0,0 +1,5 @@
+config BR2_PACKAGE_RPIC_CONTROLLER
+	bool "rpic-controller"
+	help
+	  Adds controller package
+	  https://github.com/rpicopter/controller
diff -rupN buildroot-2014.08.orig/package/rpicopter/rpic-controller/rpic-controller.mk buildroot-2014.08/package/rpicopter/rpic-controller/rpic-controller.mk
--- buildroot-2014.08.orig/package/rpicopter/rpic-controller/rpic-controller.mk	1970-01-01 01:00:00.000000000 +0100
+++ buildroot-2014.08/package/rpicopter/rpic-controller/rpic-controller.mk	2014-11-09 21:34:50.575759992 +0000
@@ -0,0 +1,22 @@
+#############################################################
+#
+# rpic-controller 
+#
+#############################################################
+RPIC_CONTROLLER_VERSION = master
+RPIC_CONTROLLER_SITE_METHOD = git
+RPIC_CONTROLLER_REPO_URL = https://github.com/rpicopter/controller.git
+RPIC_CONTROLLER_SITE = $(call qstrip,$(RPIC_CONTROLLER_REPO_URL))
+#PIGPIO_INSTALL_STAGING = YES
+
+#$(MAKE) CC=$(TARGET_CC) LD=$(TARGET_LD) -C $(@D) all
+
+define RPIC_CONTROLLER_BUILD_CMDS
+ $(MAKE) $(TARGET_CONFIGURE_OPTS) -C $(@D) all
+endef
+
+define RPIC_CONTROLLER_INSTALL_TARGET_CMDS
+	DESTDIR=$(TARGET_DIR) $(MAKE) -C $(@D) install 
+endef
+
+$(eval $(generic-package))
diff -rupN buildroot-2014.08.orig/package/rpicopter/rpicopter.mk buildroot-2014.08/package/rpicopter/rpicopter.mk
--- buildroot-2014.08.orig/package/rpicopter/rpicopter.mk	1970-01-01 01:00:00.000000000 +0100
+++ buildroot-2014.08/package/rpicopter/rpicopter.mk	2014-11-09 21:34:50.575759992 +0000
@@ -0,0 +1 @@
+include package/rpicopter/*/*.mk
diff -rupN buildroot-2014.08.orig/package/rpicopter/rpic-pigpio/Config.in buildroot-2014.08/package/rpicopter/rpic-pigpio/Config.in
--- buildroot-2014.08.orig/package/rpicopter/rpic-pigpio/Config.in	1970-01-01 01:00:00.000000000 +0100
+++ buildroot-2014.08/package/rpicopter/rpic-pigpio/Config.in	2014-11-09 21:34:50.575759992 +0000
@@ -0,0 +1,5 @@
+config BR2_PACKAGE_RPIC_PIGPIO
+	bool "rpic-pigpio"
+	help
+	  Adds pigpio package
+	  https://github.com/rpicopter/pigpio
diff -rupN buildroot-2014.08.orig/package/rpicopter/rpic-pigpio/rpic-pigpio.mk buildroot-2014.08/package/rpicopter/rpic-pigpio/rpic-pigpio.mk
--- buildroot-2014.08.orig/package/rpicopter/rpic-pigpio/rpic-pigpio.mk	1970-01-01 01:00:00.000000000 +0100
+++ buildroot-2014.08/package/rpicopter/rpic-pigpio/rpic-pigpio.mk	2014-11-09 21:34:50.575759992 +0000
@@ -0,0 +1,34 @@
+#############################################################
+#
+# rpic-pigpio 
+#
+#############################################################
+#RPIC_PIGPIO_VERSION = a26ed773d7792cca7f67c7baca530940cb991e5d
+RPIC_PIGPIO_VERSION = master
+RPIC_PIGPIO_SITE_METHOD = git
+RPIC_PIGPIO_REPO_URL = https://github.com/rpicopter/pigpio.git
+RPIC_PIGPIO_SITE = $(call qstrip,$(RPIC_PIGPIO_REPO_URL))
+#PIGPIO_INSTALL_STAGING = YES
+
+#$(MAKE) CC=$(TARGET_CC) LD=$(TARGET_LD) -C $(@D) all
+
+define RPIC_PIGPIO_BUILD_CMDS
+ $(MAKE) $(TARGET_CONFIGURE_OPTS) -C $(@D) all
+endef
+
+define RPIC_PIGPIO_INSTALL_TARGET_CMDS
+	$(INSTALL) -D -m 0755 -d              $(TARGET_DIR)/opt/pigpio/cgi
+	$(INSTALL)  -m 0755 -d              $(TARGET_DIR)/usr/local/include
+	$(INSTALL)  -m 0644 $(@D)/pigpio.h        $(TARGET_DIR)/usr/local/include
+	$(INSTALL) -m 0644 $(@D)/pigpiod_if.h    $(TARGET_DIR)/usr/local/include
+	$(INSTALL) -m 0755 -d              $(TARGET_DIR)/usr/local/lib
+	$(INSTALL) -m 0644 $(@D)/libpigpio.a     $(TARGET_DIR)/usr/local/lib
+	$(INSTALL) -m 0644 $(@D)/libpigpiod_if.a $(TARGET_DIR)/usr/local/lib
+	$(INSTALL) -m 0755 -d              $(TARGET_DIR)/usr/local/bin
+	$(INSTALL) -m 0755 $(@D)/pig2vcd         $(TARGET_DIR)/usr/local/bin
+	$(INSTALL)  -m 0755 $(@D)/pigpiod        $(TARGET_DIR)/usr/local/bin
+	$(INSTALL)  -m 0755 $(@D)/pigs           $(TARGET_DIR)/usr/local/bin
+
+endef
+
+$(eval $(generic-package))
diff -rupN buildroot-2014.08.orig/package/rpicopter/rpic-qtsixa/Config.in buildroot-2014.08/package/rpicopter/rpic-qtsixa/Config.in
--- buildroot-2014.08.orig/package/rpicopter/rpic-qtsixa/Config.in	1970-01-01 01:00:00.000000000 +0100
+++ buildroot-2014.08/package/rpicopter/rpic-qtsixa/Config.in	2014-11-09 21:34:50.575759992 +0000
@@ -0,0 +1,6 @@
+config BR2_PACKAGE_RPIC_QTSIXA
+	bool "rpic-qtsixa"
+	depends on BR2_PACKAGE_LIBUSB_COMPAT
+	help
+	  Adds support for PS3 SixAxis controller
+	  https://github.com/rpicopter/QtSixA 
diff -rupN buildroot-2014.08.orig/package/rpicopter/rpic-qtsixa/rpic-qtsixa.mk buildroot-2014.08/package/rpicopter/rpic-qtsixa/rpic-qtsixa.mk
--- buildroot-2014.08.orig/package/rpicopter/rpic-qtsixa/rpic-qtsixa.mk	1970-01-01 01:00:00.000000000 +0100
+++ buildroot-2014.08/package/rpicopter/rpic-qtsixa/rpic-qtsixa.mk	2014-11-09 21:34:50.575759992 +0000
@@ -0,0 +1,44 @@
+#############################################################
+#
+# rpic-qtsixa
+#
+#############################################################
+#QTSIXA_VERSION = ac0a6c3ef460acebd8f60fa9a44946056deed0aa
+RPIC_QTSIXA_VERSION = master
+RPIC_QTSIXA_SITE_METHOD = git
+RPIC_QTSIXA_REPO_URL = https://github.com/rpicopter/QtSixA.git
+RPIC_QTSIXA_SITE = $(call qstrip,$(RPIC_QTSIXA_REPO_URL))
+
+define RPIC_QTSIXA_BUILD_CMDS
+ $(MAKE) $(TARGET_CONFIGURE_OPTS) -C $(@D)/sixad
+ $(MAKE) $(TARGET_CONFIGURE_OPTS) -C $(@D)/utils
+endef
+
+define RPIC_QTSIXA_INSTALL_TARGET_CMDS
+       $(INSTALL) -m 0755 -d              $(TARGET_DIR)/usr/bin
+       $(INSTALL) -m 0755 -d              $(TARGET_DIR)/usr/sbin
+       $(INSTALL) -m 0755 -d              $(TARGET_DIR)/etc/default
+       $(INSTALL) -m 0755 -d              $(TARGET_DIR)/etc/init.d
+       $(INSTALL) -m 0755 -d              $(TARGET_DIR)/etc/logrotate.d
+       $(INSTALL) -m 0755 -d              $(TARGET_DIR)/var/lib/sixad
+       $(INSTALL) -m 0755 -d              $(TARGET_DIR)/var/lib/sixad/profiles
+
+	$(INSTALL) -m 644 $(@D)/sixad/sixad.default $(TARGET_DIR)/etc/default/sixad
+	$(INSTALL) -m 644 $(@D)/sixad/sixad.default $(TARGET_DIR)/etc/init.d/sixad
+	$(INSTALL) -m 644 $(@D)/sixad/sixad.log $(TARGET_DIR)/etc/logrotate.d/sixad
+	$(INSTALL) -m 755 $(@D)/sixad/sixad $(TARGET_DIR)/usr/bin/
+	$(INSTALL) -m 755 $(@D)/sixad/bins/sixad-bin $(TARGET_DIR)/usr/sbin/
+	$(INSTALL) -m 755 $(@D)/sixad/bins/sixad-sixaxis $(TARGET_DIR)/usr/sbin/
+	$(INSTALL) -m 755 $(@D)/sixad/bins/sixad-remote $(TARGET_DIR)/usr/sbin/
+	$(INSTALL) -m 755 $(@D)/sixad/bins/sixad-3in1 $(TARGET_DIR)/usr/sbin/
+	$(INSTALL) -m 755 $(@D)/sixad/bins/sixad-raw $(TARGET_DIR)/usr/sbin/
+	$(INSTALL) -m 755 $(@D)/sixad/sixad-dbus-blocker $(TARGET_DIR)/usr/sbin/
+	@chmod 777 -R $(TARGET_DIR)/var/lib/sixad/
+
+
+	$(INSTALL) -m 755 $(@D)/utils/bins/sixpair $(TARGET_DIR)/usr/local/bin/
+	$(INSTALL) -m 755 $(@D)/utils/bins/sixpair-kbd $(TARGET_DIR)/usr/local/bin/
+	$(INSTALL) -m 755 $(@D)/utils/bins/hidraw-dump $(TARGET_DIR)/usr/local/bin/
+endef
+
+$(eval $(generic-package))
diff -rupN buildroot-2014.08.orig/package/rpicopter/rpic-servoblaster/Config.in buildroot-2014.08/package/rpicopter/rpic-servoblaster/Config.in
--- buildroot-2014.08.orig/package/rpicopter/rpic-servoblaster/Config.in	1970-01-01 01:00:00.000000000 +0100
+++ buildroot-2014.08/package/rpicopter/rpic-servoblaster/Config.in	2014-11-09 21:34:50.575759992 +0000
@@ -0,0 +1,5 @@
+config BR2_PACKAGE_RPIC_SERVOBLASTER
+	bool "rpic-servoblaster"
+	help
+	  Adds servoblaster
+	  https://github.com/rpicopter/ServoBlaster
diff -rupN buildroot-2014.08.orig/package/rpicopter/rpic-servoblaster/rpic-servoblaster.mk buildroot-2014.08/package/rpicopter/rpic-servoblaster/rpic-servoblaster.mk
--- buildroot-2014.08.orig/package/rpicopter/rpic-servoblaster/rpic-servoblaster.mk	1970-01-01 01:00:00.000000000 +0100
+++ buildroot-2014.08/package/rpicopter/rpic-servoblaster/rpic-servoblaster.mk	2014-11-09 21:34:50.575759992 +0000
@@ -0,0 +1,21 @@
+#############################################################
+#
+# rpic-servoblaster
+#
+#############################################################
+#SERVOBLASTER_VERSION = ac0a6c3ef460acebd8f60fa9a44946056deed0aa
+RPIC_SERVOBLASTER_VERSION = master
+RPIC_SERVOBLASTER_SITE_METHOD = git
+RPIC_SERVOBLASTER_REPO_URL = https://github.com/rpicopter/servoblaster.git
+RPIC_SERVOBLASTER_SITE = $(call qstrip,$(RPIC_SERVOBLASTER_REPO_URL))
+
+define RPIC_SERVOBLASTER_BUILD_CMDS
+ $(MAKE) $(TARGET_CONFIGURE_OPTS) -C $(@D)/user all
+endef
+
+define RPIC_SERVOBLASTER_INSTALL_TARGET_CMDS
+       $(INSTALL) -m 0755 -d              $(TARGET_DIR)/usr/local/bin
+       $(INSTALL) -m 0755 $(@D)/user/servod         $(TARGET_DIR)/usr/local/bin
+endef
+
+$(eval $(generic-package))
diff -rupN buildroot-2014.08.orig/package/rpicopter/rpic-utils/Config.in buildroot-2014.08/package/rpicopter/rpic-utils/Config.in
--- buildroot-2014.08.orig/package/rpicopter/rpic-utils/Config.in	1970-01-01 01:00:00.000000000 +0100
+++ buildroot-2014.08/package/rpicopter/rpic-utils/Config.in	2014-11-09 21:34:50.575759992 +0000
@@ -0,0 +1,7 @@
+config BR2_PACKAGE_RPIC_UTILS
+	bool "rpic-utils"
+#	select BR2_PACKAGE_BLUEZ_UTILS_COMPAT
+#	select BR2_PACKAGE_UDEV
+	help
+	  Adds utils package
+	  https://github.com/rpicopter/utils
diff -rupN buildroot-2014.08.orig/package/rpicopter/rpic-utils/rpic-utils.mk buildroot-2014.08/package/rpicopter/rpic-utils/rpic-utils.mk
--- buildroot-2014.08.orig/package/rpicopter/rpic-utils/rpic-utils.mk	1970-01-01 01:00:00.000000000 +0100
+++ buildroot-2014.08/package/rpicopter/rpic-utils/rpic-utils.mk	2014-11-09 21:34:50.575759992 +0000
@@ -0,0 +1,22 @@
+#############################################################
+#
+# rpic-utils 
+#
+#############################################################
+RPIC_UTILS_VERSION = master
+RPIC_UTILS_SITE_METHOD = git
+RPIC_UTILS_REPO_URL = https://github.com/rpicopter/utils.git
+RPIC_UTILS_SITE = $(call qstrip,$(RPIC_UTILS_REPO_URL))
+#PIGPIO_INSTALL_STAGING = YES
+
+#$(MAKE) CC=$(TARGET_CC) LD=$(TARGET_LD) -C $(@D) all
+
+define RPIC_UTILS_BUILD_CMDS
+endef
+
+
+define RPIC_UTILS_INSTALL_TARGET_CMDS
+	DESTDIR=$(TARGET_DIR) $(MAKE) -C $(@D) install 
+endef 
+
+$(eval $(generic-package))
diff -rupN buildroot-2014.08.orig/package/rpi-firmware/cmdline.txt buildroot-2014.08/package/rpi-firmware/cmdline.txt
--- buildroot-2014.08.orig/package/rpi-firmware/cmdline.txt	2014-09-01 12:20:56.000000000 +0100
+++ buildroot-2014.08/package/rpi-firmware/cmdline.txt	2014-11-09 21:34:50.575759992 +0000
@@ -1 +1 @@
-dwc_otg.fiq_fix_enable=1 sdhci-bcm2708.sync_after_dma=0 dwc_otg.lpm_enable=0 console=tty1 root=/dev/mmcblk0p2 rootwait
+dwc_otg.fiq_fix_enable=1 sdhci-bcm2708.sync_after_dma=0 dwc_otg.lpm_enable=0 console=ttyAMA0,115200 root=/dev/mmcblk0p2 rootwait
diff -rupN buildroot-2014.08.orig/package/rpi-firmware/config.txt buildroot-2014.08/package/rpi-firmware/config.txt
--- buildroot-2014.08.orig/package/rpi-firmware/config.txt	2014-09-01 12:20:56.000000000 +0100
+++ buildroot-2014.08/package/rpi-firmware/config.txt	2014-11-09 21:34:50.575759992 +0000
@@ -1,14 +1,17 @@
-# Please note that this is only a sample, we recommend you to change it to fit
-# your needs.
-# You should override this file using a post-build script.
-# See http://buildroot.org/downloads/manual/manual.html#rootfs-custom
-# and http://elinux.org/RPiconfig for a description of config.txt syntax
+#arm_freq=900
+#core_freq=250
+#sdram_freq=450
+#over_voltage=2
 
 arm_freq=700
 core_freq=250
-kernel=zImage
-disable_overscan=1
-gpu_mem_256=100
-gpu_mem_512=100
 sdram_freq=400
 over_voltage=0
+
+kernel=zImage
+disable_overscan=1
+gpu_mem_256=128
+gpu_mem_512=256
+
+start_file=start.elf
+fixup_file=fixup.dat
diff -rupN buildroot-2014.08.orig/package/rpi-firmware/rpi-firmware.mk buildroot-2014.08/package/rpi-firmware/rpi-firmware.mk
--- buildroot-2014.08.orig/package/rpi-firmware/rpi-firmware.mk	2014-09-01 12:20:56.000000000 +0100
+++ buildroot-2014.08/package/rpi-firmware/rpi-firmware.mk	2014-11-09 21:34:50.575759992 +0000
@@ -4,7 +4,7 @@
 #
 ################################################################################
 
-RPI_FIRMWARE_VERSION = 16ebc76904699b036f709beb65ee5ff70f83190b
+RPI_FIRMWARE_VERSION = f2275b92477d1a01f45f321d648f0eeeeeb0a67e
 RPI_FIRMWARE_SITE = $(call github,raspberrypi,firmware,$(RPI_FIRMWARE_VERSION))
 RPI_FIRMWARE_LICENSE = BSD-3c
 RPI_FIRMWARE_LICENSE_FILES = boot/LICENCE.broadcom
