diff -rupN rpic-qtsixa.orig/sixad/sixad rpic-qtsixa/sixad/sixad
--- rpic-qtsixa.orig/sixad/sixad	2014-04-06 21:39:03.782835500 +0100
+++ rpic-qtsixa/sixad/sixad	2014-04-06 21:40:18.512895792 +0100
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/bin/sh
 
 # sixad-bin wrapper
 # written by falkTX
@@ -36,11 +36,11 @@ fi
 }
 
 sixad_running_check () {
-ps -e | grep sixad-bin > /dev/null
+pidof sixad-bin > /dev/null
 }
 
 bluetoothd_running_check () {
-ps -e | grep bluetoothd > /dev/null
+pidof bluetoothd > /dev/null
 }
 
 modprobe_check () {
@@ -62,7 +62,7 @@ fi
 
 bt_stop() {
 if (bluetoothd_running_check); then
-  $SUDO pkill -KILL bluetoothd
+  $SUDO killall bluetoothd
 fi
 
 $SUDO /usr/sbin/sixad-dbus-blocker &
@@ -87,9 +87,9 @@ fi
   ;;
 
   --stop|-stop|stop)
-$SUDO pkill -KILL sixad-sixaxis
-$SUDO pkill -KILL sixad-remote
-$SUDO pkill -TERM sixad-bin
+$SUDO killall sixad-sixaxis
+$SUDO killall sixad-remote
+$SUDO killall sixad-bin
 bt_start
   ;;
 
