diff -rupN buildroot-2015.02.orig/package/Config.in buildroot-2015.02/package/Config.in
--- buildroot-2015.02.orig/package/Config.in	2015-03-01 21:26:12.000000000 +0000
+++ buildroot-2015.02/package/Config.in	2015-05-17 11:43:22.344219369 +0100
@@ -2,6 +2,10 @@ menu "Target packages"
 
 	source "package/busybox/Config.in"
 
+menu "Rpicopter"
+        source "package/rpicopter/Config.in"
+endmenu
+
 menu "Audio and video applications"
 	source "package/alsa-utils/Config.in"
 	source "package/aumix/Config.in"
diff -rupN buildroot-2015.02.orig/package/rpicopter/Config.in buildroot-2015.02/package/rpicopter/Config.in
--- buildroot-2015.02.orig/package/rpicopter/Config.in	1970-01-01 01:00:00.000000000 +0100
+++ buildroot-2015.02/package/rpicopter/Config.in	2015-05-17 11:43:23.016426790 +0100
@@ -0,0 +1,34 @@
+config BR2_PACKAGE_RPICOPTER
+	bool "rpicopter"
+	help
+	  Adds rpicopter utils package
+	  https://github.com/rpicopter 
+
+if BR2_PACKAGE_RPICOPTER
+source "package/rpicopter/rpic-controller/Config.in"
+source "package/rpicopter/rpic-gpac/Config.in"
+
+#source "package/rpicopter/rpic-janusgateway/Config.in"
+#source "package/rpicopter/rpic-libnice/Config.in"
+#source "package/rpicopter/rpic-sofiasip/Config.in"
+#source "package/rpicopter/rpic-dinglibs/Config.in"
+#source "package/rpicopter/rpic-libsrtp/Config.in"
+#source "package/rpicopter/rpic-libwebsock/Config.in"
+
+source "package/rpicopter/rpic-avrspi/Config.in"
+source "package/rpicopter/rpic-ps3/Config.in"
+source "package/rpicopter/rpic-baro/Config.in"
+source "package/rpicopter/rpic-gps/Config.in"
+source "package/rpicopter/rpic-utils/Config.in"
+source "package/rpicopter/rpic-www/Config.in"
+source "package/rpicopter/rpic-camera/Config.in"
+source "package/rpicopter/rpic-wilds/Config.in"
+source "package/rpicopter/rpic-gstcamsrc/Config.in"
+
+source "package/rpicopter/rpic-raspimjpeg/Config.in"
+source "package/rpicopter/rpic-bluez-tools/Config.in"
+source "package/rpicopter/rpic-utils/Config.in"
+source "package/rpicopter/rpic-pigpio/Config.in"
+source "package/rpicopter/rpic-servoblaster/Config.in"
+source "package/rpicopter/rpic-qtsixa/Config.in"
+endif
diff -rupN buildroot-2015.02.orig/package/rpicopter/rpic-avrspi/Config.in buildroot-2015.02/package/rpicopter/rpic-avrspi/Config.in
--- buildroot-2015.02.orig/package/rpicopter/rpic-avrspi/Config.in	1970-01-01 01:00:00.000000000 +0100
+++ buildroot-2015.02/package/rpicopter/rpic-avrspi/Config.in	2015-05-17 11:43:23.020428023 +0100
@@ -0,0 +1,5 @@
+config BR2_PACKAGE_RPIC_AVRSPI
+	bool "rpic-avrspi"
+	help
+	  Adds avrspi controller package
+	  https://github.com/rpicopter/AvrMiniCopter-AvrSpi
diff -rupN buildroot-2015.02.orig/package/rpicopter/rpic-avrspi/rpic-avrspi.mk buildroot-2015.02/package/rpicopter/rpic-avrspi/rpic-avrspi.mk
--- buildroot-2015.02.orig/package/rpicopter/rpic-avrspi/rpic-avrspi.mk	1970-01-01 01:00:00.000000000 +0100
+++ buildroot-2015.02/package/rpicopter/rpic-avrspi/rpic-avrspi.mk	2015-05-17 11:43:23.020428023 +0100
@@ -0,0 +1,21 @@
+#############################################################
+#
+# rpic-avrspi 
+#
+#############################################################
+RPIC_AVRSPI_VERSION = master
+RPIC_AVRSPI_SITE_METHOD = git
+RPIC_AVRSPI_REPO_URL = https://github.com/rpicopter/AvrMiniCopter-AVRSPI.git
+RPIC_AVRSPI_SITE = $(call qstrip,$(RPIC_AVRSPI_REPO_URL))
+RPIC_AVRSPI_DEPENDENCIES = linux
+
+define RPIC_AVRSPI_BUILD_CMDS
+	$(MAKE) $(TARGET_CONFIGURE_OPTS) -C $(@D) all
+#	$(MAKE) -C $(@D) $(LINUX_MAKE_FLAGS) KERNEL_SRC=$(LINUX_DIR) KVERSION=$(LINUX_VERSION_PROBED) modules
+endef
+
+define RPIC_AVRSPI_INSTALL_TARGET_CMDS
+	DESTDIR=$(TARGET_DIR) $(MAKE) -C $(@D) install 
+endef
+
+$(eval $(generic-package))
diff -rupN buildroot-2015.02.orig/package/rpicopter/rpic-baro/Config.in buildroot-2015.02/package/rpicopter/rpic-baro/Config.in
--- buildroot-2015.02.orig/package/rpicopter/rpic-baro/Config.in	1970-01-01 01:00:00.000000000 +0100
+++ buildroot-2015.02/package/rpicopter/rpic-baro/Config.in	2015-05-17 11:43:23.016426790 +0100
@@ -0,0 +1,7 @@
+config BR2_PACKAGE_RPIC_BARO
+	bool "rpic-baro"
+#	select BR2_PACKAGE_BLUEZ_UTILS_COMPAT
+#	select BR2_PACKAGE_UDEV
+	help
+	  Adds basic altitude hold based on barometer to AvrMiniCopter
+	  https://github.com/rpicopter/AvrMiniCopter-Baro
diff -rupN buildroot-2015.02.orig/package/rpicopter/rpic-baro/rpic-baro.mk buildroot-2015.02/package/rpicopter/rpic-baro/rpic-baro.mk
--- buildroot-2015.02.orig/package/rpicopter/rpic-baro/rpic-baro.mk	1970-01-01 01:00:00.000000000 +0100
+++ buildroot-2015.02/package/rpicopter/rpic-baro/rpic-baro.mk	2015-05-17 11:43:23.016426790 +0100
@@ -0,0 +1,23 @@
+#############################################################
+#
+# rpic-baro 
+#
+#############################################################
+RPIC_BARO_VERSION = master
+RPIC_BARO_SITE_METHOD = git
+RPIC_BARO_REPO_URL = https://github.com/rpicopter/AvrMiniCopter-Baro.git
+RPIC_BARO_SITE = $(call qstrip,$(RPIC_BARO_REPO_URL))
+#PIGPIO_INSTALL_STAGING = YES
+
+#$(MAKE) CC=$(TARGET_CC) LD=$(TARGET_LD) -C $(@D) all
+
+define RPIC_BARO_BUILD_CMDS
+	$(MAKE) $(TARGET_CONFIGURE_OPTS) -C $(@D) all
+endef
+
+
+define RPIC_BARO_INSTALL_TARGET_CMDS
+	DESTDIR=$(TARGET_DIR) $(MAKE) -C $(@D) install 
+endef 
+
+$(eval $(generic-package))
diff -rupN buildroot-2015.02.orig/package/rpicopter/rpic-bluez-tools/Config.in buildroot-2015.02/package/rpicopter/rpic-bluez-tools/Config.in
--- buildroot-2015.02.orig/package/rpicopter/rpic-bluez-tools/Config.in	1970-01-01 01:00:00.000000000 +0100
+++ buildroot-2015.02/package/rpicopter/rpic-bluez-tools/Config.in	2015-05-17 11:43:23.016426790 +0100
@@ -0,0 +1,7 @@
+config BR2_PACKAGE_RPIC_BLUEZ_TOOLS
+    bool "rpic-bluez-tools"
+    select BR2_PACKAGE_DBUS_GLIB
+#    depends on BR2_PACKAGE_READLINE
+    help
+        bluez-tools
+        https://github.com/rpicopter/bluez-tools
diff -rupN buildroot-2015.02.orig/package/rpicopter/rpic-bluez-tools/rpic-bluez-tools.mk buildroot-2015.02/package/rpicopter/rpic-bluez-tools/rpic-bluez-tools.mk
--- buildroot-2015.02.orig/package/rpicopter/rpic-bluez-tools/rpic-bluez-tools.mk	1970-01-01 01:00:00.000000000 +0100
+++ buildroot-2015.02/package/rpicopter/rpic-bluez-tools/rpic-bluez-tools.mk	2015-05-17 11:43:23.016426790 +0100
@@ -0,0 +1,14 @@
+#############################################################
+#
+# rpic-bluez-tools 
+#
+#############################################################
+RPIC_BLUEZ_TOOLS_VERSION = master
+RPIC_BLUEZ_TOOLS_DEPENDENCIES = dbus-glib
+RPIC_BLUEZ_TOOLS_SITE_METHOD = git
+RPIC_BLUEZ_TOOLS_REPO_URL = https://github.com/rpicopter/bluez-tools.git
+RPIC_BLUEZ_TOOLS_SITE = $(call qstrip,$(RPIC_BLUEZ_TOOLS_REPO_URL))
+RPIC_BLUEZ_TOOLS_INSTALL_TARGET = YES
+RPIC_BLUEZ_TOOLS_AUTORECONF = YES
+
+$(eval $(autotools-package))
diff -rupN buildroot-2015.02.orig/package/rpicopter/rpic-camera/Config.in buildroot-2015.02/package/rpicopter/rpic-camera/Config.in
--- buildroot-2015.02.orig/package/rpicopter/rpic-camera/Config.in	1970-01-01 01:00:00.000000000 +0100
+++ buildroot-2015.02/package/rpicopter/rpic-camera/Config.in	2015-05-17 11:43:23.024429257 +0100
@@ -0,0 +1,7 @@
+config BR2_PACKAGE_RPIC_CAMERA
+	bool "rpic-camera"
+#	select BR2_PACKAGE_BLUEZ_UTILS_COMPAT
+#	select BR2_PACKAGE_UDEV
+	help
+	  Adds camera (FPV) support for AvrMiniCopter
+	  https://github.com/rpicopter/AvrMiniCopter-Camera
diff -rupN buildroot-2015.02.orig/package/rpicopter/rpic-camera/rpic-camera.mk buildroot-2015.02/package/rpicopter/rpic-camera/rpic-camera.mk
--- buildroot-2015.02.orig/package/rpicopter/rpic-camera/rpic-camera.mk	1970-01-01 01:00:00.000000000 +0100
+++ buildroot-2015.02/package/rpicopter/rpic-camera/rpic-camera.mk	2015-05-17 11:43:23.024429257 +0100
@@ -0,0 +1,20 @@
+#############################################################
+#
+# rpic-camera 
+#
+#############################################################
+RPIC_CAMERA_VERSION = master
+RPIC_CAMERA_SITE_METHOD = git
+RPIC_CAMERA_REPO_URL = https://github.com/rpicopter/AvrMiniCopter-Camera.git
+RPIC_CAMERA_SITE = $(call qstrip,$(RPIC_CAMERA_REPO_URL))
+
+define RPIC_CAMERA_BUILD_CMDS
+	$(MAKE) $(TARGET_CONFIGURE_OPTS) -C $(@D) all
+endef
+
+
+define RPIC_CAMERA_INSTALL_TARGET_CMDS
+	DESTDIR=$(TARGET_DIR) $(MAKE) -C $(@D) install 
+endef 
+
+$(eval $(generic-package))
diff -rupN buildroot-2015.02.orig/package/rpicopter/rpic-controller/Config.in buildroot-2015.02/package/rpicopter/rpic-controller/Config.in
--- buildroot-2015.02.orig/package/rpicopter/rpic-controller/Config.in	1970-01-01 01:00:00.000000000 +0100
+++ buildroot-2015.02/package/rpicopter/rpic-controller/Config.in	2015-05-17 11:43:23.024429257 +0100
@@ -0,0 +1,5 @@
+config BR2_PACKAGE_RPIC_CONTROLLER
+	bool "rpic-controller"
+	help
+	  Adds controller package
+	  https://github.com/rpicopter/controller
diff -rupN buildroot-2015.02.orig/package/rpicopter/rpic-controller/rpic-controller.mk buildroot-2015.02/package/rpicopter/rpic-controller/rpic-controller.mk
--- buildroot-2015.02.orig/package/rpicopter/rpic-controller/rpic-controller.mk	1970-01-01 01:00:00.000000000 +0100
+++ buildroot-2015.02/package/rpicopter/rpic-controller/rpic-controller.mk	2015-05-17 11:43:23.024429257 +0100
@@ -0,0 +1,22 @@
+#############################################################
+#
+# rpic-controller 
+#
+#############################################################
+RPIC_CONTROLLER_VERSION = master
+RPIC_CONTROLLER_SITE_METHOD = git
+RPIC_CONTROLLER_REPO_URL = https://github.com/rpicopter/controller.git
+RPIC_CONTROLLER_SITE = $(call qstrip,$(RPIC_CONTROLLER_REPO_URL))
+#PIGPIO_INSTALL_STAGING = YES
+
+#$(MAKE) CC=$(TARGET_CC) LD=$(TARGET_LD) -C $(@D) all
+
+define RPIC_CONTROLLER_BUILD_CMDS
+ $(MAKE) $(TARGET_CONFIGURE_OPTS) -C $(@D) all
+endef
+
+define RPIC_CONTROLLER_INSTALL_TARGET_CMDS
+	DESTDIR=$(TARGET_DIR) $(MAKE) -C $(@D) install 
+endef
+
+$(eval $(generic-package))
diff -rupN buildroot-2015.02.orig/package/rpicopter/rpic-gpac/Config.in buildroot-2015.02/package/rpicopter/rpic-gpac/Config.in
--- buildroot-2015.02.orig/package/rpicopter/rpic-gpac/Config.in	1970-01-01 01:00:00.000000000 +0100
+++ buildroot-2015.02/package/rpicopter/rpic-gpac/Config.in	2015-05-17 11:43:23.016426790 +0100
@@ -0,0 +1,6 @@
+config BR2_PACKAGE_RPIC_GPAC
+	bool "rpic-gpac"
+	help
+	Adds multimedia framework GPAC
+	https://github.com/gpac/gpac
+	
diff -rupN buildroot-2015.02.orig/package/rpicopter/rpic-gpac/rpic-gpac.mk buildroot-2015.02/package/rpicopter/rpic-gpac/rpic-gpac.mk
--- buildroot-2015.02.orig/package/rpicopter/rpic-gpac/rpic-gpac.mk	1970-01-01 01:00:00.000000000 +0100
+++ buildroot-2015.02/package/rpicopter/rpic-gpac/rpic-gpac.mk	2015-05-17 11:43:23.016426790 +0100
@@ -0,0 +1,36 @@
+#############################################################
+#
+# rpic-gpac 
+#
+#############################################################
+RPIC_GPAC_VERSION = master
+RPIC_GPAC_SITE_METHOD = git
+RPIC_GPAC_REPO_URL = https://github.com/gpac/gpac.git 
+RPIC_GPAC_SITE = $(call qstrip,$(RPIC_GPAC_REPO_URL))
+
+RPIC_GPAC_OPT += $(call qstrip,$(BR2_PACKAGE_RPIC_GPAC_EXTRACONF))
+
+
+RPIC_GPAC_CONF_OPTS = \
+        --prefix=/usr 
+
+define RPIC_GPAC_CONFIGURE_CMDS
+        (cd $(RPIC_GPAC_SRCDIR) && rm -rf config.cache && \
+        $(TARGET_CONFIGURE_OPTS) \
+        $(TARGET_CONFIGURE_ARGS) \
+        $(RPIC_GPAC_CONF_ENV) \
+        ./configure \
+                --enable-cross-compile  \
+                --cross-prefix=$(TARGET_CROSS) \
+                --sysroot=$(STAGING_DIR) \
+                --host-cc="$(HOSTCC)" \
+                --arch=$(BR2_ARCH) \
+		--cpu=$(BR2_GCC_TARGET_TUNE) \
+                --target-os="linux" \
+                --disable-stripping \
+                $(SHARED_STATIC_LIBS_OPTS) \
+                $(RPIC_GPAC_CONF_OPTS) \
+        )
+endef
+
+$(eval $(autotools-package))
diff -rupN buildroot-2015.02.orig/package/rpicopter/rpic-gps/Config.in buildroot-2015.02/package/rpicopter/rpic-gps/Config.in
--- buildroot-2015.02.orig/package/rpicopter/rpic-gps/Config.in	1970-01-01 01:00:00.000000000 +0100
+++ buildroot-2015.02/package/rpicopter/rpic-gps/Config.in	2015-05-17 11:43:23.016426790 +0100
@@ -0,0 +1,5 @@
+config BR2_PACKAGE_RPIC_GPS
+	bool "rpic-gps"
+	help
+	  To be provided
+	  https://github.com/rpicopter/AvrMiniCopter-GPS
diff -rupN buildroot-2015.02.orig/package/rpicopter/rpic-gps/rpic-gps.mk buildroot-2015.02/package/rpicopter/rpic-gps/rpic-gps.mk
--- buildroot-2015.02.orig/package/rpicopter/rpic-gps/rpic-gps.mk	1970-01-01 01:00:00.000000000 +0100
+++ buildroot-2015.02/package/rpicopter/rpic-gps/rpic-gps.mk	2015-05-17 11:43:23.016426790 +0100
@@ -0,0 +1,23 @@
+#############################################################
+#
+# rpic-gps 
+#
+#############################################################
+RPIC_GPS_VERSION = master
+RPIC_GPS_SITE_METHOD = git
+RPIC_GPS_REPO_URL = https://github.com/rpicopter/AvrMiniCopter-GPS.git
+RPIC_GPS_SITE = $(call qstrip,$(RPIC_GPS_REPO_URL))
+#PIGPIO_INSTALL_STAGING = YES
+
+#$(MAKE) CC=$(TARGET_CC) LD=$(TARGET_LD) -C $(@D) all
+
+define RPIC_GPS_BUILD_CMDS
+	$(MAKE) $(TARGET_CONFIGURE_OPTS) -C $(@D) all
+endef
+
+
+define RPIC_GPS_INSTALL_TARGET_CMDS
+	DESTDIR=$(TARGET_DIR) $(MAKE) -C $(@D) install 
+endef 
+
+$(eval $(generic-package))
diff -rupN buildroot-2015.02.orig/package/rpicopter/rpic-gstcamsrc/Config.in buildroot-2015.02/package/rpicopter/rpic-gstcamsrc/Config.in
--- buildroot-2015.02.orig/package/rpicopter/rpic-gstcamsrc/Config.in	1970-01-01 01:00:00.000000000 +0100
+++ buildroot-2015.02/package/rpicopter/rpic-gstcamsrc/Config.in	2015-05-17 11:43:23.020428023 +0100
@@ -0,0 +1,4 @@
+config BR2_PACKAGE_RPIC_GSTCAMSRC
+	bool "rpic-gstcamsrc"
+	help
+		Adds https://github.com/thaytan/gst-rpicamsrc 
diff -rupN buildroot-2015.02.orig/package/rpicopter/rpic-gstcamsrc/rpic-gstcamsrc.mk buildroot-2015.02/package/rpicopter/rpic-gstcamsrc/rpic-gstcamsrc.mk
--- buildroot-2015.02.orig/package/rpicopter/rpic-gstcamsrc/rpic-gstcamsrc.mk	1970-01-01 01:00:00.000000000 +0100
+++ buildroot-2015.02/package/rpicopter/rpic-gstcamsrc/rpic-gstcamsrc.mk	2015-05-17 11:43:23.020428023 +0100
@@ -0,0 +1,28 @@
+#############################################################
+#
+# rpic-gstcamsrc
+#
+#############################################################
+RPIC_GSTCAMSRC_VERSION = master
+RPIC_GSTCAMSRC_SITE_METHOD = git
+RPIC_GSTCAMSRC_REPO_URL = https://github.com/thaytan/gst-rpicamsrc.git 
+RPIC_GSTCAMSRC_SITE = $(call qstrip,$(RPIC_GSTCAMSRC_REPO_URL))
+RPIC_GSTCAMSRC_DEPENDENCIES = rpi-userland
+
+#RPIC_GSTCAMSRC_BCM_H = $(TARGET_DIR)/usr/include/interface/ 
+RPIC_GSTCAMSRC_BCM_H = -I$(STAGING_DIR)/usr/include/interface -I$(STAGING_DIR)/usr/include/interface/vcos/pthreads/ -I$(STAGING_DIR)/usr/include/interface/vmcs_host/linux 
+
+RPIC_GSTCAMSRC_AUTORECONF = YES
+
+RPIC_GSTCAMSRC_CONF_OPTS = INCLUDEPATH="$(RPIC_GSTCAMSRC_BCM_H)" LT_SYS_LIBRARY_PATH="/usr/lib"
+
+define RPIC_GSTCAMSRC_BUILD_CMDS
+	$(MAKE) $(TARGET_CONFIGURE_OPTS) INCLUDEPATH=$(RPIC_GSTCAMSRC_BCM_H) -C $(@D) all
+endef
+
+define RPIC_GSTCAMSRC_INSTALL_TARGET_CMDS
+	DESTDIR=$(TARGET_DIR) $(MAKE) -C $(@D) install 
+endef
+
+$(eval $(autotools-package))
+#$(eval $(generic-package))
diff -rupN buildroot-2015.02.orig/package/rpicopter/rpicopter.mk buildroot-2015.02/package/rpicopter/rpicopter.mk
--- buildroot-2015.02.orig/package/rpicopter/rpicopter.mk	1970-01-01 01:00:00.000000000 +0100
+++ buildroot-2015.02/package/rpicopter/rpicopter.mk	2015-05-17 11:43:23.012425555 +0100
@@ -0,0 +1 @@
+include package/rpicopter/*/*.mk
diff -rupN buildroot-2015.02.orig/package/rpicopter/rpic-pigpio/Config.in buildroot-2015.02/package/rpicopter/rpic-pigpio/Config.in
--- buildroot-2015.02.orig/package/rpicopter/rpic-pigpio/Config.in	1970-01-01 01:00:00.000000000 +0100
+++ buildroot-2015.02/package/rpicopter/rpic-pigpio/Config.in	2015-05-17 11:43:23.016426790 +0100
@@ -0,0 +1,5 @@
+config BR2_PACKAGE_RPIC_PIGPIO
+	bool "rpic-pigpio"
+	help
+	  Adds pigpio package
+	  https://github.com/rpicopter/pigpio
diff -rupN buildroot-2015.02.orig/package/rpicopter/rpic-pigpio/rpic-pigpio.mk buildroot-2015.02/package/rpicopter/rpic-pigpio/rpic-pigpio.mk
--- buildroot-2015.02.orig/package/rpicopter/rpic-pigpio/rpic-pigpio.mk	1970-01-01 01:00:00.000000000 +0100
+++ buildroot-2015.02/package/rpicopter/rpic-pigpio/rpic-pigpio.mk	2015-05-17 11:43:23.016426790 +0100
@@ -0,0 +1,34 @@
+#############################################################
+#
+# rpic-pigpio 
+#
+#############################################################
+#RPIC_PIGPIO_VERSION = a26ed773d7792cca7f67c7baca530940cb991e5d
+RPIC_PIGPIO_VERSION = master
+RPIC_PIGPIO_SITE_METHOD = git
+RPIC_PIGPIO_REPO_URL = https://github.com/rpicopter/pigpio.git
+RPIC_PIGPIO_SITE = $(call qstrip,$(RPIC_PIGPIO_REPO_URL))
+#PIGPIO_INSTALL_STAGING = YES
+
+#$(MAKE) CC=$(TARGET_CC) LD=$(TARGET_LD) -C $(@D) all
+
+define RPIC_PIGPIO_BUILD_CMDS
+ $(MAKE) $(TARGET_CONFIGURE_OPTS) -C $(@D) all
+endef
+
+define RPIC_PIGPIO_INSTALL_TARGET_CMDS
+	$(INSTALL) -D -m 0755 -d              $(TARGET_DIR)/opt/pigpio/cgi
+	$(INSTALL)  -m 0755 -d              $(TARGET_DIR)/usr/local/include
+	$(INSTALL)  -m 0644 $(@D)/pigpio.h        $(TARGET_DIR)/usr/local/include
+	$(INSTALL) -m 0644 $(@D)/pigpiod_if.h    $(TARGET_DIR)/usr/local/include
+	$(INSTALL) -m 0755 -d              $(TARGET_DIR)/usr/local/lib
+	$(INSTALL) -m 0644 $(@D)/libpigpio.a     $(TARGET_DIR)/usr/local/lib
+	$(INSTALL) -m 0644 $(@D)/libpigpiod_if.a $(TARGET_DIR)/usr/local/lib
+	$(INSTALL) -m 0755 -d              $(TARGET_DIR)/usr/local/bin
+	$(INSTALL) -m 0755 $(@D)/pig2vcd         $(TARGET_DIR)/usr/local/bin
+	$(INSTALL)  -m 0755 $(@D)/pigpiod        $(TARGET_DIR)/usr/local/bin
+	$(INSTALL)  -m 0755 $(@D)/pigs           $(TARGET_DIR)/usr/local/bin
+
+endef
+
+$(eval $(generic-package))
diff -rupN buildroot-2015.02.orig/package/rpicopter/rpic-ps3/Config.in buildroot-2015.02/package/rpicopter/rpic-ps3/Config.in
--- buildroot-2015.02.orig/package/rpicopter/rpic-ps3/Config.in	1970-01-01 01:00:00.000000000 +0100
+++ buildroot-2015.02/package/rpicopter/rpic-ps3/Config.in	2015-05-17 11:43:23.012425555 +0100
@@ -0,0 +1,5 @@
+config BR2_PACKAGE_RPIC_PS3
+	bool "rpic-ps3"
+	help
+	  Adds PS3 controller support over bluetooth
+	  https://github.com/rpicopter/avrcontroller-ps3
diff -rupN buildroot-2015.02.orig/package/rpicopter/rpic-ps3/rpic-ps3.mk buildroot-2015.02/package/rpicopter/rpic-ps3/rpic-ps3.mk
--- buildroot-2015.02.orig/package/rpicopter/rpic-ps3/rpic-ps3.mk	1970-01-01 01:00:00.000000000 +0100
+++ buildroot-2015.02/package/rpicopter/rpic-ps3/rpic-ps3.mk	2015-05-17 11:43:23.012425555 +0100
@@ -0,0 +1,20 @@
+#############################################################
+#
+# rpic-ps3 
+#
+#############################################################
+RPIC_PS3_VERSION = master
+RPIC_PS3_SITE_METHOD = git
+RPIC_PS3_REPO_URL = https://github.com/rpicopter/AvrMiniCopter-PS3.git
+RPIC_PS3_SITE = $(call qstrip,$(RPIC_PS3_REPO_URL))
+RPIC_PS3_DEPENDENCIES = linux
+
+define RPIC_PS3_BUILD_CMDS
+	$(MAKE) $(TARGET_CONFIGURE_OPTS) -C $(@D) all
+endef
+
+define RPIC_PS3_INSTALL_TARGET_CMDS
+	DESTDIR=$(TARGET_DIR) $(MAKE) -C $(@D) install 
+endef
+
+$(eval $(generic-package))
diff -rupN buildroot-2015.02.orig/package/rpicopter/rpic-qtsixa/Config.in buildroot-2015.02/package/rpicopter/rpic-qtsixa/Config.in
--- buildroot-2015.02.orig/package/rpicopter/rpic-qtsixa/Config.in	1970-01-01 01:00:00.000000000 +0100
+++ buildroot-2015.02/package/rpicopter/rpic-qtsixa/Config.in	2015-05-17 11:43:23.016426790 +0100
@@ -0,0 +1,6 @@
+config BR2_PACKAGE_RPIC_QTSIXA
+	bool "rpic-qtsixa"
+	depends on BR2_PACKAGE_LIBUSB_COMPAT
+	help
+	  Adds support for PS3 SixAxis controller
+	  https://github.com/rpicopter/QtSixA 
diff -rupN buildroot-2015.02.orig/package/rpicopter/rpic-qtsixa/rpic-qtsixa.mk buildroot-2015.02/package/rpicopter/rpic-qtsixa/rpic-qtsixa.mk
--- buildroot-2015.02.orig/package/rpicopter/rpic-qtsixa/rpic-qtsixa.mk	1970-01-01 01:00:00.000000000 +0100
+++ buildroot-2015.02/package/rpicopter/rpic-qtsixa/rpic-qtsixa.mk	2015-05-17 11:43:23.016426790 +0100
@@ -0,0 +1,45 @@
+#############################################################
+#
+# rpic-qtsixa
+#
+#############################################################
+#QTSIXA_VERSION = ac0a6c3ef460acebd8f60fa9a44946056deed0aa
+RPIC_QTSIXA_VERSION = master
+RPIC_QTSIXA_SITE_METHOD = git
+RPIC_QTSIXA_DEPENDENCIES = bluez_utils
+RPIC_QTSIXA_REPO_URL = https://github.com/rpicopter/QtSixA.git
+RPIC_QTSIXA_SITE = $(call qstrip,$(RPIC_QTSIXA_REPO_URL))
+
+define RPIC_QTSIXA_BUILD_CMDS
+ $(MAKE) $(TARGET_CONFIGURE_OPTS) -C $(@D)/sixad
+ $(MAKE) $(TARGET_CONFIGURE_OPTS) -C $(@D)/utils
+endef
+
+define RPIC_QTSIXA_INSTALL_TARGET_CMDS
+       $(INSTALL) -m 0755 -d              $(TARGET_DIR)/usr/bin
+       $(INSTALL) -m 0755 -d              $(TARGET_DIR)/usr/sbin
+       $(INSTALL) -m 0755 -d              $(TARGET_DIR)/etc/default
+       $(INSTALL) -m 0755 -d              $(TARGET_DIR)/etc/init.d
+       $(INSTALL) -m 0755 -d              $(TARGET_DIR)/etc/logrotate.d
+       $(INSTALL) -m 0755 -d              $(TARGET_DIR)/var/lib/sixad
+       $(INSTALL) -m 0755 -d              $(TARGET_DIR)/var/lib/sixad/profiles
+
+	$(INSTALL) -m 644 $(@D)/sixad/sixad.default $(TARGET_DIR)/etc/default/sixad
+	$(INSTALL) -m 644 $(@D)/sixad/sixad.default $(TARGET_DIR)/etc/init.d/sixad
+	$(INSTALL) -m 644 $(@D)/sixad/sixad.log $(TARGET_DIR)/etc/logrotate.d/sixad
+	$(INSTALL) -m 755 $(@D)/sixad/sixad $(TARGET_DIR)/usr/bin/
+	$(INSTALL) -m 755 $(@D)/sixad/bins/sixad-bin $(TARGET_DIR)/usr/sbin/
+	$(INSTALL) -m 755 $(@D)/sixad/bins/sixad-sixaxis $(TARGET_DIR)/usr/sbin/
+	$(INSTALL) -m 755 $(@D)/sixad/bins/sixad-remote $(TARGET_DIR)/usr/sbin/
+	$(INSTALL) -m 755 $(@D)/sixad/bins/sixad-3in1 $(TARGET_DIR)/usr/sbin/
+	$(INSTALL) -m 755 $(@D)/sixad/bins/sixad-raw $(TARGET_DIR)/usr/sbin/
+	$(INSTALL) -m 755 $(@D)/sixad/sixad-dbus-blocker $(TARGET_DIR)/usr/sbin/
+	@chmod 777 -R $(TARGET_DIR)/var/lib/sixad/
+
+
+	$(INSTALL) -m 755 $(@D)/utils/bins/sixpair $(TARGET_DIR)/usr/local/bin/
+	$(INSTALL) -m 755 $(@D)/utils/bins/sixpair-kbd $(TARGET_DIR)/usr/local/bin/
+	$(INSTALL) -m 755 $(@D)/utils/bins/hidraw-dump $(TARGET_DIR)/usr/local/bin/
+endef
+
+$(eval $(generic-package))
diff -rupN buildroot-2015.02.orig/package/rpicopter/rpic-raspimjpeg/Config.in buildroot-2015.02/package/rpicopter/rpic-raspimjpeg/Config.in
--- buildroot-2015.02.orig/package/rpicopter/rpic-raspimjpeg/Config.in	1970-01-01 01:00:00.000000000 +0100
+++ buildroot-2015.02/package/rpicopter/rpic-raspimjpeg/Config.in	2015-05-17 11:43:23.016426790 +0100
@@ -0,0 +1,4 @@
+config BR2_PACKAGE_RPIC_RASPIMJPEG
+	bool "rpic-raspimjpeg"
+	help
+		Adds raspimjpeg from https://github.com/rpicopter/raspimjpeg
diff -rupN buildroot-2015.02.orig/package/rpicopter/rpic-raspimjpeg/rpic-raspimjpeg.mk buildroot-2015.02/package/rpicopter/rpic-raspimjpeg/rpic-raspimjpeg.mk
--- buildroot-2015.02.orig/package/rpicopter/rpic-raspimjpeg/rpic-raspimjpeg.mk	1970-01-01 01:00:00.000000000 +0100
+++ buildroot-2015.02/package/rpicopter/rpic-raspimjpeg/rpic-raspimjpeg.mk	2015-05-17 11:43:23.016426790 +0100
@@ -0,0 +1,23 @@
+#############################################################
+#
+# rpic-raspimjpeg
+#
+#############################################################
+RPIC_RASPIMJPEG_VERSION = master
+RPIC_RASPIMJPEG_SITE_METHOD = git
+RPIC_RASPIMJPEG_REPO_URL = https://github.com/rpicopter/raspimjpeg.git
+RPIC_RASPIMJPEG_SITE = $(call qstrip,$(RPIC_RASPIMJPEG_REPO_URL))
+RPIC_RASPIMJPEG_DEPENDENCIES = rpi-userland rpic-gpac
+
+RPIC_RASPIMJPEG_BCM_H = $(TARGET_DIR)/usr/include/interface/ 
+RPIC_RASPIMJPEG_BCM_H = $(STAGING_DIR)/usr/include/interface/ 
+
+define RPIC_RASPIMJPEG_BUILD_CMDS
+	$(MAKE) $(TARGET_CONFIGURE_OPTS) BCM_H_DIR=$(RPIC_RASPIMJPEG_BCM_H) -C $(@D) all
+endef
+
+define RPIC_RASPIMJPEG_INSTALL_TARGET_CMDS
+	DESTDIR=$(TARGET_DIR) $(MAKE) -C $(@D) install 
+endef
+
+$(eval $(generic-package))
diff -rupN buildroot-2015.02.orig/package/rpicopter/rpic-servoblaster/Config.in buildroot-2015.02/package/rpicopter/rpic-servoblaster/Config.in
--- buildroot-2015.02.orig/package/rpicopter/rpic-servoblaster/Config.in	1970-01-01 01:00:00.000000000 +0100
+++ buildroot-2015.02/package/rpicopter/rpic-servoblaster/Config.in	2015-05-17 11:43:23.016426790 +0100
@@ -0,0 +1,5 @@
+config BR2_PACKAGE_RPIC_SERVOBLASTER
+	bool "rpic-servoblaster"
+	help
+	  Adds servoblaster
+	  https://github.com/rpicopter/ServoBlaster
diff -rupN buildroot-2015.02.orig/package/rpicopter/rpic-servoblaster/rpic-servoblaster.mk buildroot-2015.02/package/rpicopter/rpic-servoblaster/rpic-servoblaster.mk
--- buildroot-2015.02.orig/package/rpicopter/rpic-servoblaster/rpic-servoblaster.mk	1970-01-01 01:00:00.000000000 +0100
+++ buildroot-2015.02/package/rpicopter/rpic-servoblaster/rpic-servoblaster.mk	2015-05-17 11:43:23.016426790 +0100
@@ -0,0 +1,21 @@
+#############################################################
+#
+# rpic-servoblaster
+#
+#############################################################
+#SERVOBLASTER_VERSION = ac0a6c3ef460acebd8f60fa9a44946056deed0aa
+RPIC_SERVOBLASTER_VERSION = master
+RPIC_SERVOBLASTER_SITE_METHOD = git
+RPIC_SERVOBLASTER_REPO_URL = https://github.com/rpicopter/servoblaster.git
+RPIC_SERVOBLASTER_SITE = $(call qstrip,$(RPIC_SERVOBLASTER_REPO_URL))
+
+define RPIC_SERVOBLASTER_BUILD_CMDS
+ $(MAKE) $(TARGET_CONFIGURE_OPTS) -C $(@D)/user all
+endef
+
+define RPIC_SERVOBLASTER_INSTALL_TARGET_CMDS
+       $(INSTALL) -m 0755 -d              $(TARGET_DIR)/usr/local/bin
+       $(INSTALL) -m 0755 $(@D)/user/servod         $(TARGET_DIR)/usr/local/bin
+endef
+
+$(eval $(generic-package))
diff -rupN buildroot-2015.02.orig/package/rpicopter/rpic-test/Config.in buildroot-2015.02/package/rpicopter/rpic-test/Config.in
--- buildroot-2015.02.orig/package/rpicopter/rpic-test/Config.in	1970-01-01 01:00:00.000000000 +0100
+++ buildroot-2015.02/package/rpicopter/rpic-test/Config.in	2015-05-17 11:43:23.016426790 +0100
@@ -0,0 +1,4 @@
+config BR2_PACKAGE_RPIC_TEST
+	bool "rpic-test"
+	help
+	  To be provided
diff -rupN buildroot-2015.02.orig/package/rpicopter/rpic-test/rpic-test.mk buildroot-2015.02/package/rpicopter/rpic-test/rpic-test.mk
--- buildroot-2015.02.orig/package/rpicopter/rpic-test/rpic-test.mk	1970-01-01 01:00:00.000000000 +0100
+++ buildroot-2015.02/package/rpicopter/rpic-test/rpic-test.mk	2015-05-17 11:43:23.016426790 +0100
@@ -0,0 +1,24 @@
+#############################################################
+#
+# rpic-test 
+#
+#############################################################
+RPIC_TEST_VERSION = b83513f79f15
+RPIC_TEST_SITE_METHOD = hg 
+RPIC_TEST_SOURCE = wifibroadcast-$(RPIC_TEST_VERSION).zip
+RPIC_TEST_SOURCE = $(RPIC_TEST_VERSION).zip
+RPIC_TEST_SITE = https://bitbucket.org/befi/wifibroadcast/get 
+#PIGPIO_INSTALL_STAGING = YES
+
+#$(MAKE) CC=$(TARGET_CC) LD=$(TARGET_LD) -C $(@D) all
+
+define RPIC_TEST_BUILD_CMDS
+	$(MAKE) $(TARGET_CONFIGURE_OPTS) -C $(@D) all
+endef
+
+
+define RPIC_TEST_INSTALL_TARGET_CMDS
+#	DESTDIR=$(TARGET_DIR) $(MAKE) -C $(@D) install 
+endef 
+
+$(eval $(generic-package))
diff -rupN buildroot-2015.02.orig/package/rpicopter/rpic-utils/Config.in buildroot-2015.02/package/rpicopter/rpic-utils/Config.in
--- buildroot-2015.02.orig/package/rpicopter/rpic-utils/Config.in	1970-01-01 01:00:00.000000000 +0100
+++ buildroot-2015.02/package/rpicopter/rpic-utils/Config.in	2015-05-17 11:43:23.020428023 +0100
@@ -0,0 +1,7 @@
+config BR2_PACKAGE_RPIC_UTILS
+	bool "rpic-utils"
+#	select BR2_PACKAGE_BLUEZ_UTILS_COMPAT
+#	select BR2_PACKAGE_UDEV
+	help
+	  Adds utils package
+	  https://github.com/rpicopter/AvrMiniCopter-Utils
diff -rupN buildroot-2015.02.orig/package/rpicopter/rpic-utils/rpic-utils.mk buildroot-2015.02/package/rpicopter/rpic-utils/rpic-utils.mk
--- buildroot-2015.02.orig/package/rpicopter/rpic-utils/rpic-utils.mk	1970-01-01 01:00:00.000000000 +0100
+++ buildroot-2015.02/package/rpicopter/rpic-utils/rpic-utils.mk	2015-05-17 11:43:23.020428023 +0100
@@ -0,0 +1,22 @@
+#############################################################
+#
+# rpic-utils 
+#
+#############################################################
+RPIC_UTILS_VERSION = master
+RPIC_UTILS_SITE_METHOD = git
+RPIC_UTILS_REPO_URL = https://github.com/rpicopter/AvrMiniCopter-Utils.git
+RPIC_UTILS_SITE = $(call qstrip,$(RPIC_UTILS_REPO_URL))
+RPIC_UTILS_DEPENDENCIES = php wpa_supplicant
+
+#$(MAKE) CC=$(TARGET_CC) LD=$(TARGET_LD) -C $(@D) all
+
+define RPIC_UTILS_BUILD_CMDS
+endef
+
+
+define RPIC_UTILS_INSTALL_TARGET_CMDS
+	DESTDIR=$(TARGET_DIR) $(MAKE) -C $(@D) install 
+endef 
+
+$(eval $(generic-package))
diff -rupN buildroot-2015.02.orig/package/rpicopter/rpic-wilds/Config.in buildroot-2015.02/package/rpicopter/rpic-wilds/Config.in
--- buildroot-2015.02.orig/package/rpicopter/rpic-wilds/Config.in	1970-01-01 01:00:00.000000000 +0100
+++ buildroot-2015.02/package/rpicopter/rpic-wilds/Config.in	2015-05-17 11:43:23.020428023 +0100
@@ -0,0 +1,5 @@
+config BR2_PACKAGE_RPIC_WILDS
+	bool "rpic-wilds"
+	help
+	  Adds Wilds controller support
+	  https://github.com/rpicopter/AvrMiniCopter-UDP
diff -rupN buildroot-2015.02.orig/package/rpicopter/rpic-wilds/rpic-wilds.mk buildroot-2015.02/package/rpicopter/rpic-wilds/rpic-wilds.mk
--- buildroot-2015.02.orig/package/rpicopter/rpic-wilds/rpic-wilds.mk	1970-01-01 01:00:00.000000000 +0100
+++ buildroot-2015.02/package/rpicopter/rpic-wilds/rpic-wilds.mk	2015-05-17 11:43:23.020428023 +0100
@@ -0,0 +1,28 @@
+#############################################################
+#
+# rpic-wilds 
+#
+#############################################################
+RPIC_WILDS_VERSION = master
+RPIC_WILDS_SITE_METHOD = git
+RPIC_WILDS_REPO_URL = https://github.com/rpicopter/AvrMiniCopter-UDP.git 
+RPIC_WILDS_SITE = $(call qstrip,$(RPIC_WILDS_REPO_URL))
+RPIC_WILDS_DEPENDENCIES = linux
+
+define RPIC_WILDS_BUILD_CMDS
+	$(MAKE) $(TARGET_CONFIGURE_OPTS) -C $(@D) all
+endef
+
+define RPIC_WILDS_INSTALL_TARGET_CMDS
+	DESTDIR=$(TARGET_DIR) $(MAKE) -C $(@D) install 
+endef
+
+define RPIC_WILDS_INSTALL_INIT_SYSV
+        $(INSTALL) -m 0755 -D package/rpicopter/rpic-wilds/S99udpcontroller \
+                $(TARGET_DIR)/etc/init.d/S99udpcontroller
+        $(INSTALL) -m 0755 -D package/rpicopter/rpic-wilds/udpcontroller.sh \
+                $(TARGET_DIR)/usr/local/bin/udpcontroller.sh
+	$(RM) $(TARGET_DIR)/etc/init.d/udpcontroller
+endef
+
+$(eval $(generic-package))
diff -rupN buildroot-2015.02.orig/package/rpicopter/rpic-wilds/S99udpcontroller buildroot-2015.02/package/rpicopter/rpic-wilds/S99udpcontroller
--- buildroot-2015.02.orig/package/rpicopter/rpic-wilds/S99udpcontroller	1970-01-01 01:00:00.000000000 +0100
+++ buildroot-2015.02/package/rpicopter/rpic-wilds/S99udpcontroller	2015-05-17 11:43:23.020428023 +0100
@@ -0,0 +1,7 @@
+#!/bin/sh
+if [ "$1" == "start"  ]; then
+
+/usr/local/bin/udpcontroller.sh &
+
+fi
+exit 0;
diff -rupN buildroot-2015.02.orig/package/rpicopter/rpic-wilds/udpcontroller.sh buildroot-2015.02/package/rpicopter/rpic-wilds/udpcontroller.sh
--- buildroot-2015.02.orig/package/rpicopter/rpic-wilds/udpcontroller.sh	1970-01-01 01:00:00.000000000 +0100
+++ buildroot-2015.02/package/rpicopter/rpic-wilds/udpcontroller.sh	2015-05-17 11:43:23.024429257 +0100
@@ -0,0 +1,7 @@
+#!/bin/sh
+while [ 1 ]; do
+sleep 2;
+/usr/local/bin/udpcontroller
+
+done
+
diff -rupN buildroot-2015.02.orig/package/rpicopter/rpic-www/Config.in buildroot-2015.02/package/rpicopter/rpic-www/Config.in
--- buildroot-2015.02.orig/package/rpicopter/rpic-www/Config.in	1970-01-01 01:00:00.000000000 +0100
+++ buildroot-2015.02/package/rpicopter/rpic-www/Config.in	2015-05-17 11:43:23.020428023 +0100
@@ -0,0 +1,7 @@
+config BR2_PACKAGE_RPIC_WWW
+	bool "rpic-www"
+#	select BR2_PACKAGE_BLUEZ_UTILS_COMPAT
+#	select BR2_PACKAGE_UDEV
+	help
+	  Adds www package 
+	  https://github.com/rpicopter/AvrMiniCopter-WWW
diff -rupN buildroot-2015.02.orig/package/rpicopter/rpic-www/rpic-www.mk buildroot-2015.02/package/rpicopter/rpic-www/rpic-www.mk
--- buildroot-2015.02.orig/package/rpicopter/rpic-www/rpic-www.mk	1970-01-01 01:00:00.000000000 +0100
+++ buildroot-2015.02/package/rpicopter/rpic-www/rpic-www.mk	2015-05-17 11:43:23.020428023 +0100
@@ -0,0 +1,22 @@
+#############################################################
+#
+# rpic-www 
+#
+#############################################################
+RPIC_WWW_VERSION = master
+RPIC_WWW_SITE_METHOD = git
+RPIC_WWW_REPO_URL = https://github.com/rpicopter/AvrMiniCopter-WWW.git
+RPIC_WWW_SITE = $(call qstrip,$(RPIC_WWW_REPO_URL))
+#PIGPIO_INSTALL_STAGING = YES
+
+#$(MAKE) CC=$(TARGET_CC) LD=$(TARGET_LD) -C $(@D) all
+
+define RPIC_WWW_BUILD_CMDS
+endef
+
+
+define RPIC_WWW_INSTALL_TARGET_CMDS
+	DESTDIR=$(TARGET_DIR) $(MAKE) -C $(@D) install 
+endef 
+
+$(eval $(generic-package))
